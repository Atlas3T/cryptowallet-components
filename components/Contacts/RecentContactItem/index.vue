<template>
  <div class="contact-list-item">
    <div @click.prevent="goToContact">
      <div class="circle" />
      {{ contact.displayName }}
    </div>
  </div>
</template>

<script>
export default {
  name: 'RecentContactItem',
  props: {
    contact: {
      type: Object,
      required: true,
    },
  },
  methods: {
    goToContact() {
      this.$router.push({ path: `/contact-item/${this.contact.id}` });
    },
  },
};
</script>

<style scoped>
.contact-list-item > div {
  padding-right: 2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  white-space: nowrap;
}

.circle {
  background: #ececec;
  border-radius: 100%;
  height: 2.5rem;
  width: 2.5rem;
  margin-top: 0.1rem;
  margin-bottom: 0.5rem;
}
</style>
