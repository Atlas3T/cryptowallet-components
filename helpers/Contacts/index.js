import Vue from 'vue';

/**
 * Handles the contacts import and saves them in the store
 */
export default function Contacts() {
  this.vm = new Vue();
  this.import = (contacts) => {
    this.contacts = contacts;
    this.vm.$once('contactsImported', () => { return this.success(); });
    this.vm.$once('contactsImportFailure', () => { return this.failure(); });
    this.vm.$once('contactsImportError', (event) => { return this.error(event.payload); });

    try {
      const options = new ContactFindOptions();
      options.multiple = true;
      const fields = ['*'];
      navigator.contacts.find(fields, (contactsArray) => {
        this.contacts.contacts = contactsArray;
        this.vm.$emit('contactsImported');
      }, () => {
        this.vm.$emit('contactsImportFailure');
      }, options);
    } catch (err) {
      this.vm.$emit('contactsImportError', { payload: err });
    }

    return this;
  };

  this.on = (event, callback) => {
    this[event] = callback;
    return this;
  };
}
