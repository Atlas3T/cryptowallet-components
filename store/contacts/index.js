export default {
  namespaced: true,
  state: {
    contacts: [
      {
        id: 1,
        displayName: 'Terry Tibbs',
        address: 'ncousohfodicwjdclwdnc',
      },
    ],
  },
};
