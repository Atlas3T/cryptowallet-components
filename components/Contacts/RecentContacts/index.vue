<template>
  <section
    v-show="!isSearchingContacts"
    class="recent-contacts-list"
  >
    <h1 class="section-h1">
      {{ $t('recent') }}
    </h1>
    <div class="flex-scroll-area-wrapper">
      <div class="height-fix">
        <q-scroll-area class="recent-contacts-scroll-area">
          <div class="scroll-content">
            <RecentContactItem
              v-for="contact in recentContacts"
              :key="contact.name"
              :contact="contact"
            />
          </div>
        </q-scroll-area>
      </div>
    </div>
  </section>
</template>

<script>
import { mapState } from 'vuex';
import RecentContactItem from '@/components/Contacts/RecentContactItem';

export default {
  name: 'RecentContacts',
  components: {
    RecentContactItem,
  },
  computed: {
    ...mapState({
      isSearchingContacts: (state) => { return state.search.isSearchingContacts; },
      recentContacts: (state) => { return state.wallet.recentContacts; },
    }),
  },
};
</script>

<style scoped>
.height-fix {
  position: absolute;
  width: 100%;
  height: 100%;
}

.section-h1 {
  color: white;
  font-size: 0.9rem;
  font-family: 'Montserrat-Light';
  background: #162f46;
  padding: 0.5rem 1rem;
  border-bottom: 1px solid #0a2338;
  letter-spacing: 0.05rem;
  text-transform: uppercase;
}

.flex-scroll-area-wrapper {
  flex: auto;
  flex-grow: 1;
  position: relative;
  height: 100%;
}

.recent-contacts-scroll-area {
  display: flex;
  height: 100%;
}

.recent-contacts-list {
  border-top: 2px solid #09233a;
  position: relative;
  background: #16324a;
  display: flex;
  flex-direction: column;
  transition: height 100ms ease-in-out;
  min-height: 9rem;
}

.scroll-content {
  padding: 0 1rem;
  display: flex;
  height: 100%;
  align-items: center;
}
</style>
