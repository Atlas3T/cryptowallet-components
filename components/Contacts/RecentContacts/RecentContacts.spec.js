import { shallowMount } from '@vue/test-utils';
import RecentContacts from '@/components/Contacts/RecentContacts';
import RecentContactItem from '@/components/Contacts/RecentContactItem';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';

describe('RecentContacts.vue', () => {
  let storeMocks;
  let wrapper;
  let store;

  function wrapperInit (options) {
    return shallowMount(RecentContacts, options);
  }

  function storeInit (custom) {
    storeMocks = createStoreMocks(custom);
    wrapper = wrapperInit({ i18n, localVue, store: storeMocks.store });
    store = wrapper.vm.$store;
  }

  beforeEach(() => storeInit());

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a section with .recent-contacts-list class', () => {
    expect(wrapper.contains('section.recent-contacts-list')).toBe(true);
  });

  it('is hidden if store has search.isSearchingContacts === true', () => {
    expect(store.state.search.isSearchingContacts).toBe(false);
    expect(wrapper.isVisible()).toBe(true);
    store.state.search.isSearchingContacts = true
    expect(wrapper.isVisible()).toBe(false);
  });

  it('renders a h1 with .section-h1 class that contains "Recent" string', () => {
    expect(wrapper.find('h1.section-h1').html().includes('Recent')).toBe(true);
  });

  it('loops through the data and renders multiple RecentContactItem components', () => {
    expect(wrapper.findAll(RecentContactItem).length).toBeGreaterThan(1);
  });
});
