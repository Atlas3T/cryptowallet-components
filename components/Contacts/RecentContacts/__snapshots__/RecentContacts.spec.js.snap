// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentContacts.vue renders and matches snapshot 1`] = `
<section
  class="recent-contacts-list"
>
  <h1
    class="section-h1"
  >
    Recent
  </h1>
   
  <div
    class="flex-scroll-area-wrapper"
  >
    <div
      class="height-fix"
    >
      <qscrollarea-stub
        class="recent-contacts-scroll-area"
        contentactivestyle="[object Object]"
        contentstyle="[object Object]"
        delay="1000"
        thumbstyle="[object Object]"
      >
        <div
          class="scroll-content"
        >
          <recentcontactitem-stub
            contact="[object Object]"
          />
          <recentcontactitem-stub
            contact="[object Object]"
          />
          <recentcontactitem-stub
            contact="[object Object]"
          />
        </div>
      </qscrollarea-stub>
    </div>
  </div>
</section>
`;
